File: app/src/main/java/com/medo/tweetspie/MainApp.java
package com.medo.tweetspie;

import android.app.Application;
import android.support.annotation.NonNull;

import com.medo.tweetspie.database.RealmInteractor;
import com.medo.tweetspie.injection.components.AppComponent;
import com.medo.tweetspie.injection.components.DaggerAppComponent;
import com.medo.tweetspie.injection.modules.AppModule;
import com.medo.tweetspie.rest.TwitterInteractor;
import com.medo.tweetspie.system.SystemModule;
import com.medo.tweetspie.utils.CrashReportingTree;

import timber.log.Timber;


public class MainApp extends Application {

  private static AppComponent appComponent;

  @NonNull
  public static AppComponent getAppComponent() {

    return appComponent;
  }

  @Override
  public void onCreate() {

    super.onCreate();

    // dependency injection (app scope)
    appComponent = DaggerAppComponent.builder()
            .appModule(new AppModule(this))
            .systemModule(new SystemModule())
            .build();

    // init timber
    Timber.plant(BuildConfig.DEBUG ? new Timber.DebugTree() : new CrashReportingTree());

    // init twitter
    TwitterInteractor.init(this);

    // init realm
    RealmInteractor.init(this);
  }
}

File: app/src/main/java/com/medo/tweetspie/MainApp.kt
package com.medo.tweetspie

import android.app.Application
import android.util.Log
import com.medo.tweetspie.di.mainModule
import com.medo.tweetspie.storage.di.storageModule
import com.medo.tweetspie.system.di.systemModule
import com.medo.tweetspie.util.di.utilModule
import com.twitter.sdk.android.core.DefaultLogger
import com.twitter.sdk.android.core.Twitter
import com.twitter.sdk.android.core.TwitterAuthConfig
import com.twitter.sdk.android.core.TwitterConfig
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin


class MainApp : Application() {

    override fun onCreate() {
        super.onCreate()
        // init DI
        startKoin {
            androidContext(this@MainApp)
            modules(listOf(mainModule, systemModule, utilModule, storageModule))
        }
        // init twitter
        val config = TwitterConfig.Builder(this)
            .logger(DefaultLogger(Log.DEBUG))
            .twitterAuthConfig(
                TwitterAuthConfig(
                    BuildConfig.TWITTER_KEY,
                    BuildConfig.TWITTER_SECRET
                )
            )
            .debug(true)
            .build()
        Twitter.initialize(config)
    }
}
